internal/nix/search.go

```diff
@@ -0,0 +1,103 @@
+package nix
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/pkg/errors"
+	"github.com/samber/lo"
+	"go.jetpack.io/devbox/internal/debug"
+	"go.jetpack.io/devbox/internal/lock"
+)
+
+var ErrPackageNotFound = errors.New("package not found")
+var ErrPackageNotInstalled = errors.New("package not installed")
+
+func PkgExists(pkg string, lock *lock.File) (bool, error) {
+	return InputFromString(pkg, lock).validateExists()
+}
+
+type Info struct {
+	// attribute key is different in flakes vs legacy so we should only use it
+	// if we know exactly which version we are using
+	attributeKey string
+	PName        string
+	Version      string
+}
+
+func (i *Info) String() string {
+	return fmt.Sprintf("%s-%s", i.PName, i.Version)
+}
+
+func PkgInfo(nixpkgsCommit, pkg string) *Info {
+	exactPackage := fmt.Sprintf("%s#%s", FlakeNixpkgs(nixpkgsCommit), pkg)
+	if nixpkgsCommit == "" {
+		exactPackage = fmt.Sprintf("nixpkgs#%s", pkg)
```

Comment:
should we modify `FlakeNixpkgs` to:
```
func FlakeNixpkgs(commit string) string {
  if commit == "" {
     return "nixpkgs"
  }
  return "nixpkgs/" + commit
}
```
?

Reply:
@savil for sure. This is not new code though, I just moved it to other file. (and will likely get removed/changed very soon)